src/libstapsdt.c

```diff
@@ -18,6 +19,19 @@
 #include "libstapsdt.h"
 #include "errors.h"
 
+#ifdef LIBSTAPSDT_MEMORY_BACKED_FD
+#include <linux/memfd.h>
+#include <sys/mman.h>
+#include <sys/syscall.h>
+
+#define F_SEAL_SEAL 0x0001 /* prevent further seals from being set */
+
+// Note that linux must be 3.17 or greater to support this
+static inline int memfd_create(const char *name, unsigned int flags) {
+  return syscall(__NR_memfd_create, name, flags);
```

Comment:
should probbaly check that `__NR_memfd_create` is defined or else bail out elegantly.